### PR TITLE
Constraint Assignments

### PR DESCRIPTION
**Pull request by Domily11**
The relative intention view previously didn't display the constraint created already, therefore was unable to change the constraint properties. It also previously didn't include the evolving links in the options.

### CODE REVIEW

File: leaf-ui/rappid-extensions/AssignmentsTable.js [hunk 1]
```diff
@@ -237,38 +232,43 @@ var RelativeIntentionView = Backbone.View.extend({
...
     },
 
     /**
      * Loads all function segment transitions into the options dropdown menu
      */
     loadOptions: function () {
-        this.graph.getElements().filter(element => element instanceof joint.shapes.basic.Intention).forEach(intentionCell => {
-            intentionBBM = intentionCell.get('intention')
+        graph.getElements().filter(element => element instanceof joint.shapes.basic.Intention).forEach(intentionCell => {
+            var intentionBBM = intentionCell.get('intention')
```
Comment (Manarnz) [inline leaf-ui/rappid-extensions/AssignmentsTable.js]: Add `;`

File: leaf-ui/rappid-extensions/AssignmentsTable.js [hunk 2]
```diff
@@ -237,38 +232,43 @@ var RelativeIntentionView = Backbone.View.extend({
...
+        })
     },
 
     /**
      * Updates the constraint model's source id and reference time point 
      * To match the selected option
      */
     changeEpoch1: function () {
-        selectedOption = $('.epoch1List option:selected');;
+        selectedOption = this.$('.epoch1List option:selected');;
```
Comment (Manarnz) [inline leaf-ui/rappid-extensions/AssignmentsTable.js]: Remove extra `;`

File: leaf-ui/rappid-extensions/AssignmentsTable.js [hunk 3]
```diff
@@ -302,8 +302,42 @@ var RelativeIntentionView = Backbone.View.extend({
     /**
      * Helper function to create individual option dropdown elements
      */
-    getOptionTag(intentionId, nodeName, epoch) {
-        return '<option class=' + intentionId + ' epoch=' + epoch + '>' + nodeName + ': ' + epoch + '</option>';
+    getOptionTag(Id, nodeName, epoch, position) {
+        var option;
+        if (position ==='src'){
+            if (this.model.get('srcID') === Id && this.model.get('srcRefTP') === epoch){
+                //console.log("selected src " + nodeName + " " + epoch)
```
Comment (Manarnz) [inline leaf-ui/rappid-extensions/AssignmentsTable.js]: Delete `console.log`

File: leaf-ui/rappid-extensions/AssignmentsTable.js [hunk 4]
```diff
@@ -302,8 +302,42 @@ var RelativeIntentionView = Backbone.View.extend({
...
+                option = '<option class=' + Id + ' epoch="' + epoch + '" selected>' + nodeName + ': ' + epoch + '</option>';
+            }else {
```
Comment (Manarnz) [inline leaf-ui/rappid-extensions/AssignmentsTable.js]: space before `else`

File: leaf-ui/rappid-extensions/AssignmentsTable.js [hunk 5]
```diff
@@ -302,8 +302,42 @@ var RelativeIntentionView = Backbone.View.extend({
...
+                option = '<option class=' + Id + ' epoch=' + epoch + '>' + nodeName + ': ' + epoch + '</option>';
+            }
+        } else {
+            if (this.model.get('destID') === Id && this.model.get('destRefTP') === epoch){
+                option = '<option class=' + Id + ' epoch="' + epoch + '" selected>' + nodeName + ': ' + epoch + '</option>';
+            }else {
```
Comment (Manarnz) [inline leaf-ui/rappid-extensions/AssignmentsTable.js]: space before `else`

File: leaf-ui/rappid-extensions/AssignmentsTable.js [hunk 6]
```diff
@@ -302,8 +302,42 @@ var RelativeIntentionView = Backbone.View.extend({
...
+                option = '<option class=' + Id + ' epoch=' + epoch + '>' + nodeName + ': ' + epoch + '</option>';
+            }
+        }
+        return option
```
Comment (Manarnz) [inline leaf-ui/rappid-extensions/AssignmentsTable.js]: add `;`

File: leaf-ui/rappid-extensions/AssignmentsTable.js [hunk 7]
```diff
@@ -302,8 +302,42 @@ var RelativeIntentionView = Backbone.View.extend({
...
+    },
+
+    getLinkOptionTag(Id, nodeName, epoch, position) {
+        var option;
+        if (position ==='src'){
+            if (this.model.get('srcID') === Id){
+                //console.log("selected src " + nodeName + " " + epoch)
+                option = '<option class=' + Id + ' epoch="' + epoch + '" selected>' + nodeName + '</option>';
+            }else {
```
Comment (Manarnz) [inline leaf-ui/rappid-extensions/AssignmentsTable.js]: space before else

File: leaf-ui/rappid-extensions/AssignmentsTable.js [hunk 8]
```diff
@@ -302,8 +302,42 @@ var RelativeIntentionView = Backbone.View.extend({
...
+                option = '<option class=' + Id + ' epoch=' + epoch + '>' + nodeName + '</option>';
+            }
+        } else {
+            if (this.model.get('destID') === Id){
+                option = '<option class=' + Id + ' epoch="' + epoch + '" selected>' + nodeName + '</option>';
+            }else {
```
Comment (Manarnz) [inline leaf-ui/rappid-extensions/AssignmentsTable.js]: space before else

File: leaf-ui/rappid-extensions/AssignmentsTable.js [hunk 9]
```diff
@@ -302,8 +302,42 @@ var RelativeIntentionView = Backbone.View.extend({
...
+            }
+        }
+        return option
+    },
+
+    getLinkOptionTag(Id, nodeName, epoch, position) {
+        var option;
+        if (position ==='src'){
+            if (this.model.get('srcID') === Id){
+                //console.log("selected src " + nodeName + " " + epoch)
```
Comment (Manarnz) [inline leaf-ui/rappid-extensions/AssignmentsTable.js]: Delete `console.log`

File: leaf-ui/rappid-extensions/AssignmentsTable.js [hunk 10]
```diff
@@ -302,8 +302,42 @@ var RelativeIntentionView = Backbone.View.extend({
...
+                option = '<option class=' + Id + ' epoch=' + epoch + '>' + nodeName + '</option>';
+            }
+        } else {
+            if (this.model.get('destID') === Id){
+                option = '<option class=' + Id + ' epoch="' + epoch + '" selected>' + nodeName + '</option>';
+            }else {
+                option = '<option class=' + Id + ' epoch=' + epoch + '>' + nodeName + '</option>';
+            }
+        }
+        return option
```
Comment (Manarnz) [inline leaf-ui/rappid-extensions/AssignmentsTable.js]: add `;`

**Review by Manarnz — APPROVED** [review]
Well done!

**Review by amgrubb — APPROVED** [review]
Great job.

Note: Constraints now passed as `lt` and `eq` instead of `<` and `=`.
@amgrubb Update backend appropriately.

**Pull request merged**